H2MLauncher.Core/Services/H2MLauncherService.cs

```diff
@@ -1,21 +1,44 @@
-﻿using System.Text.Json;
+﻿using System.Diagnostics;
+using System.Text.Json;
+
+using H2MLauncher.Core.Utilities;
 
 namespace H2MLauncher.Core.Services
 {
-    public class H2MLauncherService(HttpClient httpClient, IErrorHandlingService errorHandlingService)
+    public class H2MLauncherService
     {
         private const string GITHUB_REPOSITORY = "https://api.github.com/repos/Bowhza/H2M-Launcher/releases";
-        private readonly HttpClient _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
-        private readonly IErrorHandlingService _errorHandlingService = errorHandlingService ?? throw new ArgumentNullException(nameof(errorHandlingService));
-        public const string CURRENT_VERSION = "H2M-v2.0.2";
+        private const string LAUNCHER = "H2MLauncher.UI.exe";
+        private const string LAUNCHER_BACKUP = $"{LAUNCHER}.backup";
+        private readonly HttpClient _httpClient;
+        private readonly IErrorHandlingService _errorHandlingService;
+        
+        public const string CURRENT_VERSION = "H2M-v2.0.3";
+
         public string LatestKnownVersion { get; private set; } = "Unknown";
 
+        public H2MLauncherService(HttpClient httpClient, IErrorHandlingService errorHandlingService)
+        {
+            _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+            _httpClient.DefaultRequestHeaders.Add("User-Agent", "H2M-Launcher-App");
+            _errorHandlingService = errorHandlingService ?? throw new ArgumentNullException(nameof(errorHandlingService));
+        }
 
         public async Task<bool> IsLauncherUpToDateAsync(CancellationToken cancellationToken)
         {
             try
             {
-                _httpClient.DefaultRequestHeaders.Add("User-Agent", "H2M-Launcher-App");
+                // remove old version if it exists
+                if (File.Exists(LAUNCHER_BACKUP))
+                    File.Delete(LAUNCHER_BACKUP);
```

Comment:
After updating i still have the backup file, but no error logged

Reply:
That is correct, only after you start the new version will it be deleted.

Reply:
Somehow that also does not happen on my side

Reply:
That is because the new version it downloads does not have the functionality yet, so you have to drag in the same version we are testing